dev/core/src/com/google/gwt/dev/jjs/ast/JCaseStatement.java

```diff
@@ -69,7 +69,7 @@ public JBinaryOperation convertToCompareExpression(JExpression value) {
   @Override
   public void traverse(JVisitor visitor, Context ctx) {
     if (visitor.visit(this, ctx)) {
-      exprs = Collections.unmodifiableList(visitor.acceptImmutable(exprs));
+      exprs = visitor.acceptWithInsertRemoveImmutable(exprs);
```

Comment:
The implementation of that method looks so weird as well, because in the end it will use `Lists.set` or `Lists.add` if `exprs` has more than one element and corresponding modifications are requested via `JModVisitor`. Shouldn't we just call `accept(exprs)`, given it is mutable now?

Reply:
Good thinking, I'll do that instead.